scripts/gitlab/build_and_test.sh

```diff
@@ -27,39 +30,86 @@ then
     build_root=$(pwd)
 fi
 
-compiler=${COMPILER:-""}
-if [[ -z ${compiler} ]]
+conf=${CONFIGURATION:-""}
+if [[ -z ${conf} ]]
 then
-    echo "COMPILER is undefined... aborting" && exit 1
+    echo "CONFIGURATION is undefined... aborting" exit 1
 fi
 
 project_dir="$(pwd)"
-build_dir="${build_root}/build_${sys_type}_${compiler}"
+build_dir="${build_root}/build_${sys_type}_${conf}"
+install_dir="${build_root}/install_${sys_type}_${conf}"
 option=${1:-""}
 
 # Build
 if [[ "${option}" != "--test-only" ]]
 then
+    # 'conf' = toolchain__tuning
+    # 'host_config' = filter__tuning
+    #   where 'filter' can represent several toolchains
+    #   like <nvcc_10_gcc_X> covers any gcc paired with nvcc10
+    # 'toolchain' is a unique set of tools, and 'tuning' allows to have
+    # several configurations for this set, like <omptarget>.
+
+    echo "--- Configuration to match :"
+    echo "* ${conf}"
+
+    toolchain=${conf/__*/}
+    tuning=${conf/${toolchain}/}
+
+    # Find project host_configs matching the configuration
+    host_configs="$(ls host-configs/${sys_type}/ | grep "\.cmake$")"
+    echo "--- Available host_configs"
+    echo "${host_configs}"
+
+    match_count=0
+    host_config=""
+
+    # Translate file names into pattern to match the host_config
+    echo "--- Patterns"
+    for hc in ${host_configs}
+    do
+        pattern="${hc//X/.*}"
+        pattern="${pattern/.cmake/}"
+        echo "${pattern}"
+
+        if [[ -n "${tuning}" && ! "${pattern}" =~ .*${tuning}$ ]]
+        then
+            continue
+        fi
+
+        if [[ "${conf}" =~ ^${pattern}$ ]]
+        then
+            (( ++match_count ))
+            host_config="${hc}"
+            echo "-> Found Project Conf : ${host_config}"
+        fi
+    done
+
+    if (( match_count > 1 )) || (( match_count == 0 ))
+    then
+        echo "ERROR : none or multiple match(s) ..."
+        exit 1
+    fi
+
     # If building, then delete everything first
     rm -rf ${build_dir} && mkdir -p ${build_dir} && cd ${build_dir}
 
-    conf_suffix="host-configs/${sys_type}/${compiler}.cmake"
-
-    generic_conf="${project_dir}/.radiuss-ci/gitlab/conf/${conf_suffix}"
+    generic_conf="${project_dir}/.radiuss-ci/gitlab/conf/host-configs/${sys_type}/${toolchain}.cmake"
     if [[ ! -f ${generic_conf} ]]
     then
         echo "ERROR: Host-config file ${generic_conf} does not exist" && exit 1
     fi
 
-    umpire_conf="${project_dir}/${conf_suffix}"
-    if [[ ! -f ${umpire_conf} ]]
+    project_conf="${project_dir}/host-configs/${sys_type}/${host_config}"
+    if [[ ! -f ${project_conf} ]]
     then
-        echo "ERROR: Host-config file ${umpire_conf} does not exist" && exit 1
+        echo "ERROR: Host-config file ${project_conf} does not exist" && exit 1
     fi
 
     cmake \
       -C ${generic_conf} \
-      -C ${umpire_conf} \
+      -C ${project_conf} \
```

Comment:
I don't see how this will perform the two filters required for [https://github.com/LLNL/radiuss-ci/commit/f4e490e571af8341c1305362f0894856c4ff9ad4#commitcomment-38158500](https://github.com/LLNL/radiuss-ci/commit/f4e490e571af8341c1305362f0894856c4ff9ad4#commitcomment-38158500).  Won't this only overlay one of the filters?

Reply:
In `project_conf`, the suffix `conf_suffix` correspond to the filter found for the `CONFIGURATION` variable passed.

So, if you ask for `gcc_8_1_0__stdc++`, `__stdc++` will be seen as a _tuning_ suffix, and `gcc_8_1_0` will have to match a toolchain host-config file.

Then the script will look for the filter matching `gcc_8_1_0` with tuning `__stdc++`.

In Umpire there are not filter yet, there is exactly one project host-config file per toolchain. But if you decide to merge all gcc host-config files into `gcc_X.camke` and `gcc_X_stdc++.cmake`, then the script would pick-up the latter.

You may also read the "Third action" section in https://github.com/LLNL/RAJA/pull/737.